driver-core/src/main/java/com/datastax/driver/core/policies/PercentileSpeculativeExecutionPolicy.java

```diff
@@ -27,41 +23,26 @@
 
 /**
  * A policy that triggers speculative executions when the request to the current host is above a given percentile.
- * <p/>
- * This class uses a {@link PerHostPercentileTracker} that must be registered with the cluster instance:
- * <pre>
- * PerHostPercentileTracker tracker = PerHostPercentileTracker
- *     .builderWithHighestTrackableLatencyMillis(15000)
- *     .build();
- * PercentileSpeculativeExecutionPolicy policy = new PercentileSpeculativeExecutionPolicy(tracker, 99.0, 2);
- * cluster = Cluster.builder()
- *     .addContactPoint("127.0.0.1")
- *     .withSpeculativeExecutionPolicy(policy)
- *     .build();
- * cluster.register(tracker);
- * </pre>
- * You <b>must</b> register the tracker with the cluster yourself (as shown on the last line above), this class will not
- * do it itself.
- * <p/>
- * <b>This class is currently provided as a beta preview: it hasn't been extensively tested yet, and the API is still subject
- * to change.</b>
  */
-@Beta
 public class PercentileSpeculativeExecutionPolicy implements SpeculativeExecutionPolicy {
-    private final PerHostPercentileTracker percentileTracker;
+    private final PercentileTracker percentileTracker;
     private final double percentile;
     private final int maxSpeculativeExecutions;
 
     /**
      * Builds a new instance.
      *
-     * @param percentileTracker        the component that will record latencies. Note that this policy doesn't register it with the {@code Cluster},
-     *                                 you <b>must</b> do it yourself (see the code example in this class's Javadoc).
-     * @param percentile               the percentile that a request's latency must fall into to be considered slow (ex: {@code 99.0}).
-     * @param maxSpeculativeExecutions the maximum number of speculative executions that will be triggered for a given request (this does not
-     *                                 include the initial, normal request). Must be strictly positive.
+     * @param percentileTracker        the component that will record latencies. It will get
+     *                                 {@link Cluster#register(LatencyTracker) registered} with the cluster when this
+     *                                 policy initializes.
+     * @param percentile               the percentile that a request's latency must fall into to be considered slow (ex:
+     *                                 {@code 99.0}).
+     * @param maxSpeculativeExecutions the maximum number of speculative executions that will be triggered for a given
+     *                                 request (this does not include the initial, normal request). Must be strictly
+     *                                 positive.
      */
-    public PercentileSpeculativeExecutionPolicy(PerHostPercentileTracker percentileTracker, double percentile, int maxSpeculativeExecutions) {
+    public PercentileSpeculativeExecutionPolicy(PercentileTracker percentileTracker,
```

Comment:
Should we add some guidance here to use `ClusterWidePercentileTracker` over `PerHostPercentileTracker` in the general case or is that better left to the docs?